Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -157,21 +190,22 @@ CGPathRef CGPathCreateCopy(CGPathRef path) {
 @Status Interoperable
 */
 CGMutablePathRef CGPathCreateMutableCopy(CGPathRef path) {
-    if (path == NULL) {
-        return NULL;
-    }
+    RETURN_NULL_IF(!path);
 
     CGMutablePathRef mutableRet = CGPathCreateMutable();
 
     // In order to call stream and copy the contents of the original path into the
     // new copy we must close this path.
     // Otherwise the D2D calls will return that a bad state has been entered.
-    path->closePath();
+    path->ClosePath();
 
-    FAIL_FAST_IF_FAILED(path->_impl.pathGeometry->Stream(mutableRet->_impl.geometrySink.Get()));
+    if (FAILED(path->PathGeometry()->Stream(mutableRet->GeometrySink().Get()))) {
+        CGPathRelease(mutableRet);
+        return NULL;
+    }
 
-    mutableRet->_impl.currentPoint = path->_impl.currentPoint;
-    mutableRet->_impl.startingPoint = path->_impl.startingPoint;
+    mutableRet->CurrentPoint() = path->CurrentPoint();
+    mutableRet->StartingPoint() = path->StartingPoint();
```

Comment:
Nope, the new figure is assumed to be open since it would just be additional work to close/re-open it for editing.

---
In reply to: [85843401](https://github.com/Microsoft/WinObjC/pull/1255#discussion_r85843401) [](ancestors = 85843401)